src/sources/redis/list.rs

```diff
@@ -0,0 +1,111 @@
+use crate::{
+    internal_events::{RedisEventReceived, RedisReceiveEventFailed},
+    shutdown::ShutdownSignal,
+    sources::redis::{create_event, Method},
+    sources::Source,
+    Pipeline,
+};
+use futures::SinkExt;
+use redis::{aio::ConnectionManager, AsyncCommands, RedisResult};
+
+pub fn watch(
+    client: redis::Client,
+    key: String,
+    redis_key: Option<String>,
+    method: Method,
+    mut shutdown: ShutdownSignal,
+    out: Pipeline,
+) -> Source {
+    let mut out = out.sink_map_err(|error| error!(message="Error sending event.", %error));
+    match method {
+        Method::Brpop => {
+            let fut = async move {
+                trace!("Get redis connection manager.");
+                let mut conn = client
+                    .get_tokio_connection_manager()
+                    .await
+                    .expect("Failed to get redis async connection.");
+                trace!("Get redis connection manager success.");
+                loop {
+                    tokio::select! {
+                        res = brpop(&mut conn,key.as_str()) => {
+                            match res {
+                                Ok(line) => {
+                                    emit!(&RedisEventReceived {
+                                        byte_size: line.len()
+                                    });
+                                    let event = create_event(line.as_str(),key.clone(),&redis_key);
+                                    tokio::select!{
```

Comment:
@pablosichert 
thanks，I don’t know hao to replace `tokio::select! {}` whih  `and_then` and `or_else` , I also see that there is still a lot of code using this usage :

```rust
loop {
    tokio::select! {
    ……
    }
}
```

Is it possible to keep this usage and modify it later?  After all, it can work correctly at the moment